### PR TITLE
Implement getCourses and getCourse endpoint in Timetable controller

### LINKED ISSUE #77: Implement Basic Courses endpoint in Timetable Controller
**Describe the task that needs to be done.**
In the PR https://github.com/SE701-T1/frontend/pull/77 from the frontend there is two API endpoints getCourses and getCourse that is relied on.
```js
/**
 * Fetches the data of all courses in the timetable
 * @return {Promise} data of all the courses
 */
export const getCourses = async () => [
  {
    courseId: 1214,
    name: 'SOFTENG 701',
    semester: '2022 sem 1',
    studentCount: 10,
    buddyCount: 4,
    updatedTime: 1647383521,
  }
];

/**
 * Fetches the data of the course with the course id
 * @param {number} courseId id of the course to retrieve
 * @return {Promise} data of the given course
 */
export const getCourse = async (courseId) => ({
  courseId,
  name: 'SOFTENG 701',
  semester: '2022 sem 1',
  studentCount: 10,
  buddyCount: 4,
  updatedTime: 1647383521,
});
```

**Describe how a solution to your proposed task might look like (and any alternatives considered).**

Implement these endpoints

**Notes**

### PR DESCRIPTION
**Pull request by scorpionknifes**
## Description
*Please include a summary of the change and which issue number is fixed (e.g. Fixes #1). Please also include relevant motivation and context.*

Adds in to two API that frontend requires use to call to find all the courses that the user has and get course information from an ID.
https://github.com/SE701-T1/frontend/pull/77

Fixes #77 

## How has this been tested?
Integration tests with SQL files are added. unit tests are added for services.

## Screenshots
image

## Checklist
*(Leave blank if not applicable)*

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: src/main/java/com/team701/buddymatcher/domain/timetable/Course.java [hunk 1]
```diff
@@ -32,63 +32,70 @@
     @ManyToMany(fetch = FetchType.EAGER, cascade = CascadeType.PERSIST)
     @JoinTable(
             name = "COURSE_STUDENT",
-            joinColumns = @JoinColumn(name = "id"),
-            inverseJoinColumns = @JoinColumn(name = "COURSE_ID"))
+            joinColumns = @JoinColumn(name = "COURSE_ID"),
+            inverseJoinColumns = @JoinColumn(name = "USER_ID"))
```
Comment (scorpionknifes) [inline src/main/java/com/team701/buddymatcher/domain/timetable/Course.java:36]: @ff-dev-45 Fixed after writing integration tests

Comment (ff-dev-45) [reply src/main/java/com/team701/buddymatcher/domain/timetable/Course.java:36]: Thanks for picking up on this and fixing it

**Review by ff-dev-45 — APPROVED** [review]
LGTM thanks for doing all of this

**Review by dlee432 — APPROVED** [review]
LGTM, thanks CZ!

**Pull request merged**